[Bug]: AMS crashes if javalin stop timeouts during leadership transistion

### What happened?

If AMS lost the leadership and dispose itself, it stops javalin HTTP server. Howerer, if javalin stop timeouts, AMS would crash.

### Affects Versions

0.5.1

### What table formats are you seeing the problem on?

Iceberg

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

```shell
2024-07-18 18:11:40,485 ERROR [main] [io.javalin.Javalin] [] - Javalin failed to stop gracefully
java.util.concurrent.TimeoutException: null
    at org.eclipse.jetty.util.FutureCallback.get(FutureCallback.java:130) ~[jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.util.FutureCallback.get(FutureCallback.java:30) ~[jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.server.handler.AbstractHandlerContainer.doShutdown(AbstractHandlerContainer.java:175) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.server.Server.doStop(Server.java:447) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:94) ~[jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
    at io.javalin.Javalin.stop(Javalin.java:209) ~[javalin-4.6.1.jar:4.6.1]
    at com.netease.arctic.server.ArcticServiceContainer.dispose(ArcticServiceContainer.java:172) ~[amoro-ams-server-0.5.2-GDC1.4.0.jar:?]
    at com.netease.arctic.server.ArcticServiceContainer.main(ArcticServiceContainer.java:113) ~[amoro-ams-server-0.5.2-GDC1.4.0.jar:?]
2024-07-18 18:11:40,489 ERROR [main] [com.netease.arctic.server.ArcticServiceContainer] [] - AMS encountered an unknown exception, will exist
io.javalin.core.util.JavalinException: java.util.concurrent.TimeoutException
    at io.javalin.Javalin.stop(Javalin.java:213) ~[javalin-4.6.1.jar:4.6.1]
    at com.netease.arctic.server.ArcticServiceContainer.dispose(ArcticServiceContainer.java:172) ~[amoro-ams-server-0.5.2-GDC1.4.0.jar:?]
    at com.netease.arctic.server.ArcticServiceContainer.main(ArcticServiceContainer.java:113) ~[amoro-ams-server-0.5.2-GDC1.4.0.jar:?]
Caused by: java.util.concurrent.TimeoutException
    at org.eclipse.jetty.util.FutureCallback.get(FutureCallback.java:130) ~[jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.util.FutureCallback.get(FutureCallback.java:30) ~[jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.server.handler.AbstractHandlerContainer.doShutdown(AbstractHandlerContainer.java:175) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.server.Server.doStop(Server.java:447) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
    at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:94) ~[jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
    at io.javalin.Javalin.stop(Javalin.java:209) ~[javalin-4.6.1.jar:4.6.1]
    ... 2 more
```


### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmoroServiceContainer.java
+++ b/amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/AmoroServiceContainer.java
@@ -172,7 +172,11 @@ public class AmoroServiceContainer {
       optimizingServiceServer.stop();
     }
     if (httpServer != null) {
-      httpServer.stop();
+      try {
+        httpServer.close();
+      } catch (Exception e) {
+        LOG.error("Error stopping http server", e);
+      }
     }
     if (tableService != null) {
       tableService.dispose();

